.github/workflows/linux.yml

```diff
@@ -55,15 +55,23 @@ jobs:
       MESON_ARGS: "-Dallow-noblas=true -Dcpu-baseline=none -Dcpu-dispatch=none"
     strategy:
       matrix:
-        version: ["3.10", "3.11", "3.12", "3.13-dev"]
+        version: ["3.10", "3.11", "3.12", "3.13", "3.13t"]
     steps:
     - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
       with:
         submodules: recursive
         fetch-tags: true
-    - uses: actions/setup-python@f677139bbe7f9c59b41e40162b753c062f5d49a3 # v5.2.0
-      with:
-        python-version: ${{ matrix.version }}
+    - uses: astral-sh/setup-uv@v3
+    - run: |
+        uv python install ${{ matrix.version }}
+        uv venv --python ${{ matrix.version }}
+        . .venv/bin/activate
+        echo PATH=$PATH >> $GITHUB_ENV
```

Comment:
This seems like either a misunderstanding in how to use `setup-uv`, or a bug in `setup-uv`. Its [docs claim](https://github.com/astral-sh/setup-uv?tab=readme-ov-file#do-i-still-need-actionssetup-python-alongside-setup-uv) that it's a drop-in replacement for `setup-python`, and we don't need this kind of path manipulation with `setup-python`.

It's possible that the cause is trying to use a `venv`. That is not what we do in CI jobs usually, and is not necessary  / just extra overhead. Is there a reason for doing this?

Reply:
uv [doesn't yet](https://docs.astral.sh/uv/guides/install-python/#getting-started) support setting up a global python environment (see the "important" note below the spot I linked), so I either need to use a virtualenv or I need to replace direct `python` invocations with `uv run`. That means I need to update `meson_actions.yml`, which means I also need to update the linux SIMD tests, since those tests use `meson_actions.yml`. I'm poking around at doing that and hopefully that's not too bad, but I hope you see why I tried using a virtualenv first.

Reply:
So I poked at this, and it looks like `uv` really doesn't want me to directly use the global uv environment. If I try to force it to do that, I get errors like:

```
  error: The interpreter at /home/runner/.local/share/uv/python/cpython-3.12.7-linux-x86_64-gnu is externally managed, and indicates the following:
    This Python installation is managed by uv and should not be modified.
  Consider creating a virtual environment with `uv venv`.
  Error: Process completed with exit code 2.
```

See [this CI run on my fork](https://github.com/ngoldbaum/numpy/actions/runs/11387828520/job/31683355423#step:8:25).

I opened an issue to ask about this on the `setup-uv` repo: https://github.com/astral-sh/setup-uv/issues/124

But for now, I think if we want to use `setup-uv`, we're more-or-less forced to use a virtualenv. I'll try to figure out where the "Defaulting to user installation because normal site-packages is not writeable" message is coming from.

Reply:
If a virtualenv is needed, having the user mess with `$GITHUB_ENV` is pretty bad UX. `setup-pixi` does this in a nicer way, by simply adding `activate-environment: true`: https://github.com/prefix-dev/setup-pixi?tab=readme-ov-file#environment-activation. That shouldn't be too hard to implement, so I hope `setup-uv` will gain that as well.